cdap-data-fabric-tests/src/test/java/co/cask/cdap/data/stream/StreamFileSizeManagerTest.java

```diff
@@ -0,0 +1,241 @@
+import co.cask.cdap.data2.transaction.stream.leveldb.LevelDBStreamFileAdmin;
+import co.cask.cdap.data2.transaction.stream.leveldb.LevelDBStreamFileConsumerFactory;
+import co.cask.cdap.gateway.MockMetricsCollectionService;
+import co.cask.cdap.gateway.router.NettyRouter;
+import co.cask.cdap.metrics.query.MetricsQueryService;
+import co.cask.cdap.security.guice.InMemorySecurityModule;
+import co.cask.cdap.test.internal.guice.AppFabricTestModule;
+import co.cask.tephra.TransactionManager;
+import com.google.common.collect.Maps;
+import com.google.common.util.concurrent.AbstractIdleService;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Provides;
+import com.google.inject.Scopes;
+import com.google.inject.Singleton;
+import com.google.inject.name.Named;
+import com.google.inject.util.Modules;
+import org.jboss.netty.handler.codec.http.HttpMethod;
+import org.jboss.netty.handler.codec.http.HttpResponseStatus;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.URL;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ *
+ */
+public class StreamFileSizeManagerTest {
+  private static final String API_KEY = "SampleTestApiKey";
+  private static final byte[] TWO_BYTES = new byte[] { 'a', 'b' };
+
+  private static final String HOSTNAME = "127.0.0.1";
+  private static int port;
+  private static CConfiguration conf;
+
+  private static Injector injector;
+  private static NettyRouter router;
+  private static MetricsQueryService metrics;
+  private static StreamHttpService streamHttpService;
+  private static TransactionManager txService;
+  private static DatasetOpExecutor dsOpService;
+  private static DatasetService datasetService;
+  private static MockHeartbeatPublisher heartbeatPublisher;
+  private static TemporaryFolder tmpFolder = new TemporaryFolder();
+
+  @BeforeClass
+  public static void beforeClass() throws IOException {
+    tmpFolder.create();
+    conf = CConfiguration.create();
+    conf.set(Constants.Router.ADDRESS, HOSTNAME);
+    conf.setInt(Constants.Router.ROUTER_PORT, 0);
+    conf.set(Constants.CFG_LOCAL_DATA_DIR, tmpFolder.newFolder().getAbsolutePath());
+    injector = startGateway(conf);
+  }
+
+  @AfterClass
+  public static void afterClass() {
+    stopGateway(conf);
+    tmpFolder.delete();
+  }
+
+  public static Injector startGateway(final CConfiguration conf) {
+    // Set up our Guice injections
+    injector = Guice.createInjector(
```

Comment:
Well I need to start the stream HTTP service, and that's how I thought I would do. But if I can do it in another way, I will.